### PR TITLE
Add ETag-per-page feature

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Expose the per-page ETag returned by the service.
Added slice of azcore.MatchConditions to ListSettingsOptions that allows for per-page match conditions. The conditions are evaluated one per page as ordered in the slice.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/data/azappconfig/go.mod [hunk 1]
```diff
@@ -14,13 +14,14 @@ require (
...
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	golang.org/x/crypto v0.22.0 // indirect
 	golang.org/x/net v0.24.0 // indirect
 	golang.org/x/sys v0.19.0 // indirect
-	golang.org/x/text v0.14.0 // indirect
+	golang.org/x/text v0.15.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/Azure/azure-sdk-for-go/sdk/azcore => ../../azcore
```
Comment (jhendrixMSFT) [inline sdk/data/azappconfig/go.mod:27]: Dependent on feature added in https://github.com/Azure/azure-sdk-for-go/pull/22932

**Review by richardpark-msft — APPROVED** [review]

File: sdk/data/azappconfig/internal/generated/custom_client.go [hunk 2]
```diff
@@ -26,3 +32,75 @@ func (a *AzureAppConfigurationClient) Pipeline() runtime.Pipeline {
 func (a *AzureAppConfigurationClient) Tracer() tracing.Tracer {
 	return a.internal.Tracer()
 }
+
+// copy of NewGetKeyValuesPager with slice of match conditions and other tweaks
+func (client *AzureAppConfigurationClient) NewGetKeyValuesPagerWithMatchConditions(matchConditions []azcore.MatchConditions, options *AzureAppConfigurationClientGetKeyValuesOptions) *runtime.Pager[AzureAppConfigurationClientGetKeyValuesResponse] {
```
Comment (RickWinter) [inline sdk/data/azappconfig/internal/generated/custom_client.go:37]: Whew... that name is a mouthful

**Review by RickWinter — APPROVED** [review]

File: sdk/data/azappconfig/go.mod (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/data/azappconfig/go.mod]: Done.

File: sdk/data/azappconfig/internal/generated/custom_client.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/data/azappconfig/internal/generated/custom_client.go:37]: Fortunately this is all generated code that isn't exported.

**Pull request merged**